ducktape/tests/loader.py

```diff
@@ -179,17 +168,84 @@ def import_modules(self, file_list):
 
         return module_list
 
-    def get_test_classes(self, module):
-        """Return list of all test classes in the module object.
-        """
+    def expand_module(self, t_info):
+        """Return a list of TestInfo objects, one object for every 'testable unit' in t_info.module"""
+
+        test_info_list = []
+        module = t_info.module
         module_objects = module.__dict__.values()
-        return [c for c in module_objects if self.is_test_class(c)]
+        test_classes = [c for c in module_objects if self.is_test_class(c)]
+
+        for cls in test_classes:
+            test_info_list.extend(self.expand_class(TestInfo(module=module, cls=cls)))
+
+        return test_info_list
+
+    def expand_class(self, t_info):
+        """Return a list of TestInfo objects, one object for each method in t_info.cls"""
+        test_methods = []
+        for f_name in dir(t_info.cls):
+            f = getattr(t_info.cls, f_name)
+            if self.is_test_function(f):
+                test_methods.append(f)
+
+        test_info_list = []
+        for f in test_methods:
+            t = TestInfo(module=t_info.module, cls=t_info.cls, function=f)
+
+            if _expandable(f):
```

Comment:
Ah, I see. Just to be clear, the issue with other non-method fields, like if I have something like this?

```
class MyTest(Test):
    field1 = [1,2,3]
    field2 = "foobar"
```